app/src/main/java/com/example/trainingdemoapp/sharedPreference/SharedPreference.kt

```diff
@@ -0,0 +1,87 @@
+package com.example.trainingdemoapp.sharedPreference
+
+import android.content.Context
+import android.content.SharedPreferences
+import com.example.trainingdemoapp.models.UserModel
+import com.example.trainingdemoapp.utils.toJson
+import com.google.gson.Gson
+import dagger.hilt.android.qualifiers.ApplicationContext
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class SharedPreference @Inject constructor(@ApplicationContext val context: Context) {
+
+    enum class SpKeys(val key: String) {
+        FILE_NAME("Demo"),
+        IS_USER_LOGIN("isUserLogin"),
+        TOKEN("token"),
+        UID("uid"),
+        CLIENT("client"),
+        USER_DATA("user")
+    }
+
+    private val preferences =
+        context.getSharedPreferences(SpKeys.FILE_NAME.key, Context.MODE_PRIVATE)
+    private var editor: SharedPreferences.Editor = preferences.edit()
+
+    fun clearPrefs() {
+        editor.clear().apply()
+    }
+
+    private fun getSharedPreferenceStringEmptyDefault(key: String): String? {
+        return preferences.getString(key, "")
+    }
+
+    private fun putSharedPreferenceString(key: String, value: String?) {
+        editor.apply {
+            putString(key, value)
+            apply()
+        }
+    }
+
+    private fun getSharedPreferenceIntEmptyDefault(key: String): Int {
+        return preferences.getInt(key, 0)
+    }
+
+    private fun putSharedPreferenceInt(key: String, value: Int) {
+        editor.apply {
+            putInt(key, value)
+            apply()
+        }
+    }
+
+
+    var isUerLogin: Boolean
+        get() = preferences.getBoolean(SpKeys.IS_USER_LOGIN.key, false)
+        set(isUserCreated) {
+            editor.apply {
+                putBoolean(SpKeys.IS_USER_LOGIN.key, isUserCreated)
+                apply()
+            }
+        }
```

Comment:
make same put and get function for boolean and for future follow the same approach

Reply:
Done